src/phoenix/server/api/routers/auth.py

```diff
@@ -187,4 +207,83 @@ async def refresh_tokens(request: Request) -> Response:
     return response
 
 
+@router.post("/initiate-password-reset")
+async def initiate_password_reset(request: Request) -> Response:
+    sender: EmailSender = request.app.state.email_sender
+    if sender is None:
+        raise HTTPException(
+            status_code=HTTP_503_SERVICE_UNAVAILABLE,
+            detail="SMTP unavailable",
+        )
+    assert isinstance(token_expiry := request.app.state.password_reset_token_expiry, timedelta)
+    data = await request.json()
+    if not (email := data.get("email")):
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Email required",
+        )
+    async with request.app.state.db() as session:
+        user = await session.scalar(_select_active_user().filter_by(email=email))
+    if user is None:
+        return Response(status_code=HTTP_204_NO_CONTENT)
+    password_reset_token_claims = PasswordResetTokenClaims(
+        subject=UserId(user.id),
+        issued_at=datetime.now(timezone.utc),
+        expiration_time=datetime.now(timezone.utc) + token_expiry,
+    )
+    token_store: TokenStore = request.app.state.get_token_store()
+    token, _ = await token_store.create_password_reset_token(password_reset_token_claims)
+    await sender.send_password_reset_email(email, token)
+    return Response(status_code=HTTP_204_NO_CONTENT)
+
+
+@router.post("/reset-password")
+async def reset_password(request: Request) -> Response:
+    data = await request.json()
+    token_store: TokenStore = request.app.state.get_token_store()
+    if (
+        not (token := data.get("token"))
+        or not isinstance((claims := await token_store.read(token)), PasswordResetTokenClaims)
+        or not claims.expiration_time
+        or claims.expiration_time < datetime.now(timezone.utc)
+    ):
+        raise HTTPException(
+            status_code=HTTP_401_UNAUTHORIZED,
+            detail="Invalid reset token",
+        )
+    assert (user_id := claims.subject)
+    async with request.app.state.db() as session:
+        user = await session.scalar(_select_active_user().filter_by(id=int(user_id)))
+    if user is None:
+        raise HTTPException(
+            status_code=HTTP_401_UNAUTHORIZED,
+            detail="Invalid reset token",
+        )
+    if user.auth_method != enums.AuthMethod.LOCAL.value:
+        raise Conflict("Cannot modify password")
```

Comment:
This error type is intended for the GraphQL API. Can we raise an `HTTPException` instead?